src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNetIntegration.cs

```diff
@@ -27,36 +35,49 @@ public static class AdoNetIntegration
         /// <param name="mdToken">The mdToken of the original method call.</param>
         /// <returns>The value returned by the instrumented method.</returns>
         [InterceptMethod(
-            TargetAssembly = "System.Data", // .NET Framework
-            TargetType = "System.Data.Common.DbCommand",
-            TargetSignatureTypes = new[] { "System.Data.Common.DbDataReader", "System.Data.CommandBehavior" },
+            TargetAssembly = FrameworkAssembly,
+            TargetType = DbCommand,
+            TargetSignatureTypes = new[] { DbDataReader, CommandBehavior },
             TargetMinimumVersion = Major4,
             TargetMaximumVersion = Major4)]
         [InterceptMethod(
-            TargetAssembly = "System.Data.Common", // .NET Core
-            TargetType = "System.Data.Common.DbCommand",
-            TargetSignatureTypes = new[] { "System.Data.Common.DbDataReader", "System.Data.CommandBehavior" },
+            TargetAssembly = CoreAssembly,
+            TargetType = DbCommand,
+            TargetSignatureTypes = new[] { DbDataReader, CommandBehavior },
             TargetMinimumVersion = Major4,
             TargetMaximumVersion = Major4)]
         public static object ExecuteDbDataReader(object @this, int behavior, int opCode, int mdToken)
         {
             var command = (DbCommand)@this;
             var commandBehavior = (CommandBehavior)behavior;
+            var instanceType = command.GetType();
 
-            var executeReader = Emit.DynamicMethodBuilder<Func<object, CommandBehavior, object>>
-               .GetOrCreateMethodCallDelegate(
-                    command.GetType(),
-                    "ExecuteDbDataReader");
+            Func<object, CommandBehavior, object> instrumentedMethod = null;
+
+            try
+            {
+                instrumentedMethod =
+                    MethodBuilder<Func<object, CommandBehavior, object>>
+                       .Start(instanceType.Assembly, mdToken, opCode, nameof(ExecuteDbDataReader))
+                       .WithConcreteType(instanceType)
+                       .WithParameters(commandBehavior)
+                       .Build();
+            }
+            catch (Exception ex)
+            {
+                Log.ErrorException($"Error resolving {DbCommand}.{nameof(ExecuteDbDataReader)}(...)", ex);
+                throw;
+            }
 
             using (var scope = CreateScope(command))
             {
                 try
                 {
-                    return executeReader(command, commandBehavior);
+                    return instrumentedMethod(command, commandBehavior);
                 }
-                catch (Exception ex) when (scope?.Span.SetExceptionForFilter(ex) ?? false)
+                catch (Exception ex)
                 {
-                    // unreachable code
+                    scope?.Span.SetException(ex);
```

Comment:
@colin-higgins @lucaspimentel Is this the preferred way to set the exception on the span? If so, I can make sure I do this moving forward

Reply:
We discussed this and decided that the code was much clearer this way than using exception filters, and the benefits of the latter are not clear. To use exception filters we needed to add a version of `SetException()` method that always returns `false`, a coalesce operator, and a `throw` that's actually unreachable.

Compare:
```csharp
catch (Exception ex) when (scope?.Span.SetExceptionForFilter(ex) ?? false)
{
    // unreachable code
    throw;
}

/* vs */

catch (Exception ex)
{
    scope?.Span.SetException(ex);
    throw;
}
```